cast/src/lib.rs

```diff
@@ -443,6 +445,62 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn ls(&self) -> Result<String> {
+        // create a vector of addresses
+        let mut addrs: Vec<H160> = Vec::new();
+
+        // fetch the chain id
+        let chain_id = self.chain_id().await?;
+
+        // for each file in $HOME/.ethereum/keystore
+        if let Ok(home_dir) = std::env::var("HOME") {
+             let path = std::path::Path::new(&home_dir).join(".ethereum").join("keystore");
+             let entries = fs::read_dir(path)?;
```

Comment:
we want a better error message here as io errors don't provide info about the file path of a file that does not exist.